### ISSUE TITLE
Q: Clean Segment multiple times.

### ISSUE THREAD

**Issue by sibero80** [marcelrv/XiaomiRobotVacuumProtocol#44]
I'd like to emulate the app functionality to allow multiple passes for rooms.
Could this be achieved by chaining sements ids?

```
{
    "method": "app_segment_clean",
    "params": [16, 16, 16],
    "id": 6764
}
```

**Comment by gjuhasz86** [comment]
I just looked at what the app does, and it sent this request:
```
{
  "id": 6157,
  "method": "app_segment_clean",
  "params": [
    {
      "segments": [
        17
      ],
      "repeat": 3,
      "clean_order_mode": 0
    }
  ]
}
```
I own an S5 Max, but hopefully this works with other models too.

**Comment by sibero80** [comment]
Thank you! will try it this weekend and report if it works from Home Assistant.
Any clue what the clean_order_mode would do?

**Comment by gjuhasz86** [comment]
Yes, I'm fairly sure it's related to the app feature where you can define what order should it clean your rooms in case you specify multiple ones. But I haven't looked into the syntax. I assume `0` means it will just clean in any order it finds suitable. I' can also take a look over the weekend to see if I can find out any more details.

**Comment by gjuhasz86** [comment]
Ok, so apparently there's a separate pair of commands: `set_clean_sequence` and `get_clean_sequence`.
The first one is used to set the sequence, but interestingly, changing it did not change the `clean_order_mode` parameter in the `app_segment_clean` request, it was still `0`.
So it may be unused, unrelated, or has some other logic that's not well understood at this point.

**Comment by sibero80** [comment]
@gjuhasz86 , have you had any luck with the `clean_order_mode` attribute?
As a side question, do you think there is a way to also set the fan_speed when cleaning a segment?

**Comment by sibero80** [comment]
@gjuhasz86 , have you had any luck with the `clean_order_mode` attribute?
As a side question, do you think there is a way to also set the fan_speed when cleaning a segment?
